migrations/educandu-2022-09-27-01-migrate-anavis-to-media-analysis.js

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable camelcase, no-console, no-await-in-loop */
+import uniqueId from '../src/utils/unique-id.js';
+
+const getTotalPreviousPartsLength = (parts, currentPartIndex) => {
+  return parts.slice(0, currentPartIndex).reduce((acc, part) => acc + part.length, 0);
+};
+
+export default class Educandu_2022_09_27_01_migrate_anavis_to_media_analysis {
+  constructor(db) {
+    this.db = db;
+  }
+
+  sectionUp(section) {
+    section.type = 'media-analysis';
+
+    if (section.content) {
+      const partsTotalLength = getTotalPreviousPartsLength(section.content.parts, section.content.parts.length);
+
+      const mediaAnalysisContent = {
+        width: section.content.width,
+        mainTrack: {
+          name: '',
+          sourceType: section.content.media.sourceType,
+          sourceUrl: section.content.media.sourceUrl,
+          copyrightNotice: section.content.media.copyrightNotice,
+          aspectRatio: section.content.media.aspectRatio,
+          showVideo: section.content.media.kind === 'video',
+          playbackRange: [0, 1],
+          volume: 1
+        },
+        secondaryTracks: [],
+        chapters: section.content.parts.map((part, index) => {
+          const lengthUntilPart = getTotalPreviousPartsLength(section.content.parts, index);
+          const sanitizedAnnotations = part.annotations
+            .filter(annotation => annotation)
+            .map(annotation => {
+              const match = annotation.match(/^(\d+\.)/);
+              return match ? annotation.replace(match[0], match[0].replace('.', '\\.')) : annotation;
+            });
+
+          return {
+            key: uniqueId.create(),
+            startPosition: lengthUntilPart / partsTotalLength,
+            color: part.color,
+            title: part.name,
+            text: sanitizedAnnotations.join('\\\n')
```

Comment:
Did you mean `'\n\n'`? This would create an empty blank line and therefore a new paragraph.

Reply:
Actually I meant it how I wrote it, just writing the new sentence in a new line.
My intention was to merge them as sentences, but I guess you want to merge them as paragraphs.
Should I change? I have not strong preference.

Reply:
Ah, I get it, I was asking myself what the first two slashes were good for. I would actually prefer paragraphs, because the original annotations themselves might have more lines, so we can separate them more. But I also don't have a strong opinion about it.

Reply:
Sure thing, changed to join as paragraphs!